render/src/umi/mod.rs

```diff
@@ -0,0 +1,307 @@
+                    end: serde_json::from_str(attributes.get("end").unwrap()).unwrap(),
+                };
+
+                let bullet_list_depth = current_line.depth;
+                let mut current_line_index = line + 1;
+                self.fetch_next_line(&mut current_line, current_line_index);
+
+                while current_line.depth != bullet_list_depth {
+                    if current_line.depth == bullet_list_depth + 1 {
+                        // Append Element to Bullet List
+                        let block = self.read_row(current_line_index);
+                        let bullet_list_entry = match block {
+                            Block::BulletListEntry(block) => block,
+                            _ => panic!(),
+                        };
+                        bullet_list.entries.append(&mut vec![bullet_list_entry]);
+                    } else {
+                        break;
+                    }
+
+                    current_line_index += 1;
+                    if !self.fetch_next_line(&mut current_line, current_line_index) {
+                        break;
+                    }
+                }
+
+                Block::BulletList(bullet_list)
+            }
+            "BulletListEntry" => {
+                let mut bullet_list_entry = BulletListEntry {
+                    keyword: TokenKind::from(SymbolKind::from(
+                        attributes.get("keyword").unwrap().as_str(),
+                    ))
+                    .try_into()
+                    .unwrap(),
+                    heading: self.read_inlines(attributes.get("heading").unwrap().to_string()),
+                    body: vec![],
+                    start: serde_json::from_str(attributes.get("start").unwrap()).unwrap(),
+                    end: serde_json::from_str(attributes.get("end").unwrap()).unwrap(),
+                };
+
+                let bullet_list_entry_depth = current_line.depth;
+                let mut current_line_index = line + 1;
+                self.fetch_next_line(&mut current_line, current_line_index);
+
+                while current_line.depth != bullet_list_entry_depth {
+                    if current_line.depth == bullet_list_entry_depth + 1 {
+                        // Append Element to Bullet List Entry Body
+                        let block = self.read_row(current_line_index);
+                        bullet_list_entry.body.append(&mut vec![block]);
+                    } else {
+                        break;
+                    }
+
+                    current_line_index += 1;
+
+                    if !self.fetch_next_line(&mut current_line, current_line_index) {
+                        break;
+                    }
+                }
+
+                Block::BulletListEntry(bullet_list_entry)
+            }
+            &_ => panic!(),
+        }
+    }
+
+    pub fn create_um(&mut self) -> Vec<Block> {
+        self.elements.clear();
+        assert!(!self.ods.is_empty());
```

Comment:
```suggestion
        debug_assert!(!self.ods.is_empty());
        if self.ods.is_empty() {
            // TODO: set log error for empty ods content
            return Vec::new();
        }
```

Just to not panic in release mode